### PR TITLE
Confetti after order confirmed

### LINKED ISSUE #822: Add confetti after order is filled successfully
This is a bit of fun :) Probably does not need to be a priority but I am investigating this now in the hopes that it will be quick.

### PR DESCRIPTION
**Pull request by Restioson**
This is on top of #820 and should wait for that to be merged first. This is intended to add a little hype to the new position and encourage the user to celebrate by sharing on Twitter :)

Right now it is just a little puff of confetti from the tick.

This needs flutter >3.7.2 to work, otherwise it will crash.

Fixes #822.

### CODE REVIEW

**Comment by klochowicz** [comment]
It would be good to document the minimum supported flutter version in readme, and someday perhaps use something like: https://fvm.app/ to manage flutter versions

**Comment by klochowicz** [comment]
@Restioson  any chance for a short recording how it looks like? :)

**Comment by Restioson** [comment]
Hi Mariusz, I'm away until the 4th and I can try do this when I get back :)

**Comment by da-kami** [comment]
Just gave this a try locally - looks great! 

Here's a gif that shows it for closing the order (any order filled will trigger it):



Should I fix the lints and we get it merged?

File: mobile/lib/features/trade/order_submission_status_dialog.dart [hunk 1]
```diff
@@ -26,24 +27,47 @@ class OrderSubmissionStatusDialog extends StatelessWidget {
       this.insetPadding = const EdgeInsets.all(50),
       this.navigateToRoute = ""});
 
+  @override
+  State<OrderSubmissionStatusDialog> createState() => _OrderSubmissionStatusDialog();
+}
+
+class _OrderSubmissionStatusDialog extends State<OrderSubmissionStatusDialog> {
+  late final ConfettiController _confettiController;
```
Comment (da-kami) [inline mobile/lib/features/trade/order_submission_status_dialog.dart:35]: ❓Why are we defining the `ConfettiController` in the `OrderSubmissionStatusDialog` *and* the `TradeScreen`?

My understanding is that the dialog cannot be closed anymore; so I think it should be enough to define the confetti on the `OrderSubmissionStatusDialog`.

I don't think it's wrong to define it on the underlying `Trade`screen as well, but I'm not sure it's necessary.

**Review by da-kami — APPROVED** [review]
Just one question comment - not sure important; from my side this PR is ready to merge if we want to.

Note: I had some local changes that are not part of the PR:

```
        modified:   mobile/macos/Podfile.lock
        modified:   mobile/pubspec.lock
```

Note sure this is relevant though; might just be due to me having some outdated versions.

**Comment by Restioson** [comment]
It's probably a hangover from when I was still testing. Unfortunately I did
this last thing before I left on holiday (I am still away with no PC or
laptop) so I didn't have time to clear it up 😅

**Review by klochowicz — APPROVED** [review]
looks awesome, let's get this in!

**Comment by Restioson** [comment]
I'd be grateful if someone can remove the unused controller first!

**Comment by klochowicz** [comment]
> I'd be grateful if someone can remove the unused controller first!
> […](#)

awesome, I'll take care of this :)

**Pull request merged**